### PR TITLE
[FIX] stock_account : error if bom is not available

### PR DESCRIPTION
**Pull request by fmdl**
**Description of the issue/feature this PR addresses:**
Create a phantom BOM for product A, 
Sale product A.
Archive the BOM, and create a new for product A.
Valide the picking.
--> Error

Because the bom_line_id should perfect fit with the bom find with _bom_find.

Fine tuning of https://github.com/odoo/odoo/commit/827f65f70634ba7a2e4175f20e5301e514ada0a0

@amoyaux 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
OPW #2492857

**Comment by fmdl** [comment]
ping @simongoffin

**Comment by fmdl** [comment]
ping @oco-odoo

**Comment by fmdl** [comment]
cc @Whenrow

File: addons/mrp_account/models/product.py [hunk 1]
```diff
@@ -44,17 +45,24 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
...
         value = 0
-        for move in stock_moves:
-            bom_line = move.bom_line_id
-            bom_line_data = bom_lines[bom_line]
-            bom_line_qty = bom_line_data['qty']
-            value += move.product_id._compute_average_price(qty_invoiced * bom_line_qty, qty_to_invoice * bom_line_qty, move)
+        for dummy, data in bom_lines:
+            bom_line_qty = data['qty']
+            bom_line_product = data['product']
+            value += bom_line_product._compute_average_price(qty_invoiced * bom_line_qty, qty_to_invoice * bom_line_qty, stock_moves)
```
Comment (Whenrow) [inline addons/mrp_account/models/product.py:65]: If you get in there stock_move is empty right ? For the sake of understanding i would pass `False` as argument

File: addons/mrp_account/models/product.py [hunk 2]
```diff
@@ -44,17 +45,24 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
...
         if stock_moves.product_id == self:
             return super()._compute_average_price(qty_invoiced, qty_to_invoice, stock_moves)
-        bom = self.env['mrp.bom']._bom_find(product=self, company_id=stock_moves.company_id.id, bom_type='phantom')
+        if stock_moves:
+            product_moves = defaultdict(lambda: self.env['stock.move'])
+            for sm in stock_moves:
+                product_moves[sm.product_id] |= sm
+            value = 0
+            for product, moves in product_moves.items():
+                qty = sum(moves.mapped('product_qty'))
```
Comment (Whenrow) [inline addons/mrp_account/models/product.py:54]: The test condition above doesn't multiply qty_invoiced with stock_move.product_qty 
Why do you need to do it here ?

File: addons/mrp_account/models/product.py (hunk 1 above)
Comment (fmdl) [reply addons/mrp_account/models/product.py:65]: `False` doesn't work, but `self.env['stock.move']` works

**Comment by fmdl** [comment]
More complexe issue https://github.com/odoo/odoo/issues/70912

**Pull request closed without merging**